src/realm/db.cpp

```diff
@@ -2622,7 +2640,11 @@ void Transaction::end_read()
 void Transaction::do_end_read() noexcept
 {
     detach();
+
     db->release_read_lock(m_read_lock);
+    if (m_oldest_version_not_persisted)
```

Comment:
here I'd be afraid that we could end up releasing the same read_lock twice (see previous note)